mysql-test/mysql-test-run.pl

```diff
@@ -4015,11 +4015,11 @@ ($$)
 
     mark_time_used('test');
     my $expected_exit = 1;
-    foreach $proc (keys(%keep_waiting_proc)) {
+    foreach my $wait_for_proc (keys(%keep_waiting_proc)) {
       # ----------------------------------------------------
       # Was it the test program that exited
       # ----------------------------------------------------
-      if ($proc eq $test)
+      if ($wait_for_proc eq $test)
       {
```

Comment:
You could do simply `if ($keep_waiting_proc{$test})` outside of the loop, right?

Even better, don't put `$test` into `%keep_waiting_proc` at all. This will avoid all kinds of problems with trying to get a coredump of the wrong process, etc. That is, don't put `$proc` into `%keep_waiting_proc` until you've checked that `$proc` is not `$test`.

Reply:
> That is, don't put $proc into %keep_waiting_proc until you've checked that $proc is not $test.

Done